quotas/business_rules.py

```diff
@@ -163,11 +164,39 @@ class ON11(PreventDeleteIfInUse):
 
 
 @only_applicable_after("2007-12-31")
-class ON12(PreventDeleteIfInUse):
+class ON12(BusinessRule):
     """The quota order number origin cannot be deleted if it is used in a
     measure."""
 
-    in_use_check = "order_number_in_use"
+    def validate(self, order_number_origin):
+        """
+        Loop over measures that reference the same quota order number as origin.
+
+        Get all the origins linked to this measure. Loop over these origins and
+        check that there are no measures linked to the origin .
+        """
+
+        measures = measures_models.Measure.objects.approved_up_to_transaction(
+            order_number_origin.transaction,
+        )
+
+        if not measures.exists():
+            return
+
+        order_numbers = measures.filter(
+            geographical_area_id=order_number_origin.geographical_area_id,
+            order_number_id=order_number_origin.order_number_id,
+        )
```

Comment:
A comment around this section is showing as resolved, but the code looks unchanged with no response. I think the suggested change was this:
```suggestion
        order_numbers = measures.filter(
            geographical_area=order_number_origin.geographical_area,
            order_number=order_number_origin.order_number,
        )
```

Reply:
... but Gabe's question about filtering by SID is also relevant.

Reply:
I am going to favour the SID update, since it has potential for a wider catchment

Reply:
ok - getting to SID is erroring. Not sure why - falling back to object (which under the hood i expect goes on PK anyway)

Reply:
ok worked it out - referencing SID working now

Reply:
@paulpepper-trade resolve if you are ok with this update ˆ